src/lib.rs

```diff
@@ -1,10 +1,47 @@
-use anyhow;
 use rev_buf_reader::RevBufReader;
-use std::io::BufRead;
+use std::io::{BufRead, BufReader};
 use std::{fmt, fs::File, io, path::Path};
 
 use chrono::NaiveDateTime;
 
+pub struct Seeker {
+    file: String,
+    id: i32,
+    // to search forward or backward through file
+    forward: bool,
+}
+
+impl Seeker {
+    pub fn new(file: String, id: i32) -> Self {
+        Seeker {
+            file,
+            id,
+            forward: false,
+        }
+    }
+
+    pub fn forward(&mut self, arg: bool) -> &mut Self {
+        self.forward = arg;
+        self
+    }
+
+    pub fn run(&self) -> Option<String> {
+        let file = self.file.clone();
+        let id = self.id;
+        let forward = self.forward;
+
+        let line = if forward {
```

Comment:
I'm not totally happy with this implementation, as we've got duplication in the code here. But for now it will do